openedx/core/djangoapps/credit/api.py

```diff
@@ -111,6 +111,91 @@ def get_credit_requirements(course_key, namespace=None):
     ]
 
 
+def get_credit_requirement_status(course_key, username):
+    """ Retrieve the user's status for each credit requirement in the course.
+
+    Args:
+        course_key (CourseKey): The identifier for course
+        username (str): The identifier of the user
+
+    Example:
+        >>> get_credit_requirement_status("course-v1-edX-DemoX-1T2015", "john")
+
+                [
+                    {
+                        "namespace": "verification",
+                        "name": "verification",
+                        "criteria": {},
+                        "status": "satisfied",
+                    },
+                    {
+                        "namespace": "reverification",
+                        "name": "midterm",
+                        "criteria": {},
+                        "status": "Not satisfied",
+                    },
+                    {
+                        "namespace": "proctored_exam",
+                        "name": "final",
+                        "criteria": {},
+                        "status": "error",
+                    },
+                    {
+                        "namespace": "grade",
+                        "name": "grade",
+                        "criteria": {"min_grade": 0.8},
+                        "status": None,
+                    },
+                ]
+
+    Returns:
+        list of requirement statuses
+    """
+    requirements = CreditRequirement.get_course_requirements(course_key)
+    statuses = []
+    for requirement in requirements:
+        status = CreditRequirementStatus.get_status(requirement, username)
+        statuses.append({
+            "namespace": requirement.namespace,
+            "name": requirement.name,
+            "criteria": requirement.criteria,
+            "status": status.status if status else None,
+            "date": status.created.strftime('%m/%d/%Y') if status else None
+        })
+    return statuses
+
+
+def get_credit_eligibility(username):
+    eligibilities = CreditEligibility.get_user_eligibility(username)
+    user_eligibilities = {}
+    for eligibility in eligibilities:
+        user_eligibilities[unicode(eligibility.course.course_key)] = {
+            "is_eligible": True,
+            "created_at": eligibility.created,
+            "providers": [
+                {
+                    "id": provider.provider_id,
+                    "display_name": provider.display_name
+                }
+                for provider in eligibility.course.get_providers()
+            ]
+        }
+
+    return user_eligibilities
+
+
+def is_credit_course(course_key):
```

Comment:
I have done this as Will previously commented on my PR to not use model functions directly and use API instead